cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -348,6 +348,10 @@ protected void startUp() throws Exception {
         Transactions.invalidateQuietly(txClient, tx);
         throw t;
       }
+    } catch (LinkageError e) {
+      // Need to wrap LinkageError. Otherwise, listeners of this Guava Service may not be called if the initialization
+      // of the user program is missing dependencies (CDAP-2543)
```

Comment:
This can happen during `run` or `destroy` method as well, right? We should handle that as well